packages/synapse-react-client/src/utils/hooks/useUploadFileEntity/useUploadFileEntities.ts

```diff
@@ -358,10 +362,16 @@ export function useUploadFileEntities(
 
   const initiateUpload = useCallback(
     (args: InitiateUploadArgs) => {
+      if (uploadDestination == null) {
+        console.error(
+          'Upload destination was not loaded, or failed to load! Aborting upload.',
+        )
+        return
+      }
```

Comment:
Had issues where `uploadDestination` was not ready when we called `initiateUpload`. I added the callback to avoid this scenario. This should not happen now, but I added an error log just in case.